src/buf_encoder.rs

```diff
@@ -30,7 +30,11 @@ impl<const CAP: usize> BufEncoder<CAP> {
 
     /// Creates an empty `BufEncoder` that will encode bytes to hex characters in the given case.
     #[inline]
-    pub fn new(case: Case) -> Self { BufEncoder { buf: ArrayString::new(), table: case.table() } }
+    #[allow(clippy::let_unit_value)] // Allow the unit value of the const check
```

Comment:
Note: this lint is stupid. It makes something that was already explicit explicit.

Reply:
https://github.com/rust-lang/rust-clippy/issues/9080